core/src/voting_service.rs

```diff
@@ -89,15 +118,21 @@ impl VotingService {
             cluster_info,
             poh_recorder,
             UPCOMING_LEADER_FANOUT_SLOTS,
+            connection_cache.protocol(),
         );
 
         if !upcoming_leader_sockets.is_empty() {
             for tpu_vote_socket in upcoming_leader_sockets {
-                let _ = cluster_info.send_transaction(vote_op.tx(), Some(tpu_vote_socket));
+                let _ = send_vote_transaction(
+                    cluster_info,
+                    vote_op.tx(),
+                    Some(tpu_vote_socket),
+                    &connection_cache,
+                );
             }
         } else {
             // Send to our own tpu vote socket if we cannot find a leader to send to
-            let _ = cluster_info.send_transaction(vote_op.tx(), None);
+            let _ = send_vote_transaction(cluster_info, vote_op.tx(), None, &connection_cache);
```

Comment:
should we introduce a metric/counter for the errors here?

Reply:
I will address it in another PR: https://github.com/anza-xyz/agave/issues/3894

Reply:
Actually we have already metrics which can be obtained from connection cache: `connection_cache_vote_quic`